components/brave_wallet_ui/utils/style.utils.ts

```diff
@@ -26,3 +26,61 @@ export const makePaddingMixin = (defaultPadding: number | string) => css<{
   padding: ${(p) =>
     p?.padding ? sizeCssValue(p.padding) : sizeCssValue(defaultPadding)};
 `
+
+export const getDominantColorFromImageURL = (src: string) => {
+  if (!src) {
+    return undefined
+  }
+
+  // Construct new image from src.
+  let img = new Image()
+  img.src = src
+
+  // Images actual width and height.
+  const { naturalWidth, naturalHeight } = img
+
+  // Higher number will have better precision but may be slower.
+  const precision = 10
+
+  // Calculate canvas sizes.
+  const scaleRatio = Math.min(
+    precision / naturalWidth,
+    precision / naturalHeight
+  )
+  const width = Math.ceil(naturalWidth * scaleRatio)
+  const height = Math.ceil(naturalHeight * scaleRatio)
+  const area = width * height
+
+  // Construct canvas.
+  let context = document.createElement('canvas').getContext('2d')
+
+  // context, width, height and area might be null.
+  if (context && width && height && area) {
+    // Set canvase width and height.
+    context.canvas.width = width
+    context.canvas.height = height
+
+    // Draw image to canvas.
+    context.drawImage(img, 0, 0, width, height)
+
+    // Get raw image data from canvas.
+    const data = context.getImageData(0, 0, width, height).data
+    let r = 0
+    let g = 0
+    let b = 0
+
+    // Compare and find average color values
+    for (let i = 0; i < data.length; i += 4) {
+      r += data[i]
+      g += data[i + 1]
+      b += data[i + 2]
+    }
+
+    r = ~~(r / area)
+    g = ~~(g / area)
+    b = ~~(b / area)
```

Comment:
Neat! Never seen this operator used before.

Reply:
Is a faster alternative to `Math.floor()` 😎